src/audio/component.c

```diff
@@ -225,3 +225,92 @@ int comp_get_copy_limits(struct comp_dev *dev, struct comp_copy_limits *cl)
 
 	return 0;
 }
+
+/* Function overwrites PCM parameters (frame_fmt, buffer_fmt, channels, rate)
+ * with buffer parameters when specific flag is set.
+ */
+static void comp_update_params(uint32_t flag,
+			       struct sof_ipc_stream_params *params,
+			       struct comp_buffer *buffer)
+{
+	if (flag & BUFF_PARAMS_FRAME_FMT)
+		params->frame_fmt = buffer->frame_fmt;
+
+	if (flag & BUFF_PARAMS_BUFFER_FMT)
+		params->buffer_fmt = buffer->buffer_fmt;
+
+	if (flag & BUFF_PARAMS_CHANNELS)
+		params->channels = buffer->channels;
+
+	if (flag & BUFF_PARAMS_RATE)
+		params->rate = buffer->rate;
+}
+
+int comp_verify_params(struct comp_dev *dev, uint32_t flag,
+		       struct sof_ipc_stream_params *params)
+{
+	struct list_item *buffer_list;
+	struct list_item *source_list;
+	struct list_item *sink_list;
+	struct list_item *clist;
+	struct comp_buffer *sinkb;
+	struct comp_buffer *buf;
+	int dir = dev->direction;
+
+	if (!params) {
+		trace_comp_error("comp_verify_params() error: !params");
+		return -EINVAL;
+	}
+
+	source_list = comp_buffer_list(dev, PPL_DIR_UPSTREAM);
+	sink_list = comp_buffer_list(dev, PPL_DIR_DOWNSTREAM);
+
+	/* searching for endpoint component e.g. HOST, DAI, DETECT_TEST, which
+	 * has only one sink or one source buffer.
+	 */
+	if (list_is_empty(source_list) != list_is_empty(sink_list)) {
+		if (!list_is_empty(source_list))
+			buf = list_first_item(&dev->bsource_list,
+					      struct comp_buffer,
+					      sink_list);
+		else
+			buf = list_first_item(&dev->bsink_list,
+					      struct comp_buffer,
+					      source_list);
+
+		/* update specific pcm paraeter with buffer parameter if
```

Comment:
Typo?

Reply:
@paulstelian97 
Thanks, typo updated. I do not understand a bit, what you mean "to be treated separately from DAI". DAI component does not use ```comp_verify_params()```. DAI just checks whether PCM parameters that it received are compatible with its hardware parameters set during ```dai_set_config()```. Here it is:
```
static int dai_verify_params(struct comp_dev *dev,
			     struct sof_ipc_stream_params *params)
{
	struct sof_ipc_stream_params hw_params;

	dai_comp_get_hw_params(dev, &hw_params);
	if ((hw_params.rate != params->rate && hw_params.rate) ||
	    (hw_params.channels != params->channels && hw_params.channels)) {
		trace_dai_error_with_ids(dev,
					 "dai_verify_params() error: pcm params does not match hw params"
					);
		return -EINVAL;
	}

	/* set component period frames */
	component_set_period_frames(dev, params->rate);

	return 0;
}
```